python/pyspark/mllib/linalg.py

```diff
@@ -1152,9 +1156,385 @@ def sparse(numRows, numCols, colPtrs, rowIndices, values):
         return SparseMatrix(numRows, numCols, colPtrs, rowIndices, values)
 
 
+class DistributedMatrix(object):
+    """
+    Represents a distributively stored matrix backed by one or
+    more RDDs.
+
+    """
+    def numRows(self):
+        """Get or compute the number of rows."""
+        raise NotImplementedError
+
+    def numCols(self):
+        """Get or compute the number of cols."""
+        raise NotImplementedError
+
+
+class RowMatrix(DistributedMatrix):
+    """
+    Represents a row-oriented distributed Matrix with no meaningful
+    row indices.
+
+    .. note:: Experimental
+    """
+    def __init__(self, rows, numRows=0, numCols=0):
+        """
+        Create a wrapper over a Java RowMatrix.
+
+        :param rows: An RDD of Vectors.
+        """
+        if not isinstance(rows, RDD):
+            raise TypeError("rows should be an RDD object, got %s" % type(rows))
+        first = rows.first()
+        if not isinstance(first, Vector):
+            raise TypeError("rows should be an RDD of Vectors, got an RDD of %s" % type(first))
+
+        javaRowMatrix = callMLlibFunc("createRowMatrix", rows, long(numRows), int(numCols))
+        self._jrm = JavaModelWrapper(javaRowMatrix)
+        self.rows = rows
+
+    @staticmethod
+    def _from_java(javaRowMatrix):
```

Comment:
For conversion, it seems you are constructing the java_model on the scala side, deconstructing it into rows and constructing the model again in the scala side again. This might be expensive.

I defer to @mengxr for decision regarding this.